mailpoet/lib/Config/Populator.php

```diff
@@ -982,4 +983,32 @@ private function updateToUnifiedTrackingSettings() {
     }
     $this->settings->set('tracking.level', $trackingLevel);
   }
+
+  private function fixNotificationHistoryRecordsStuckAtSending() {
+    // perform once for versions below or equal to 3.99.0
+    if (version_compare((string)$this->settings->get('db_version', '3.99.1'), '3.99.0', '>')) {
+      return false;
+    }
+
+    $newsletters = $this->entityManager->getClassMetadata(NewsletterEntity::class)->getTableName();
+    $queues = $this->entityManager->getClassMetadata(SendingQueueEntity::class)->getTableName();
+    $tasks = $this->entityManager->getClassMetadata(ScheduledTaskEntity::class)->getTableName();
+
+    $this->entityManager->getConnection()->executeStatement("
+      UPDATE {$newsletters} n
+      JOIN {$queues} q ON n.id = q.newsletter_id
+      JOIN {$tasks} t ON q.task_id = t.id
+      SET n.status = :sentStatus
+      WHERE n.type = :type
+      AND n.status = :sendingStatus
+      AND t.status = :taskStatus
+    ", [
+      'type' => NewsletterEntity::TYPE_NOTIFICATION_HISTORY,
+      'sendingStatus' => NewsletterEntity::STATUS_SENDING,
+      'sentStatus' => NewsletterEntity::STATUS_SENT,
+      'taskStatus' => ScheduledTaskEntity::STATUS_COMPLETED,
```

Comment:
Just a note that the notification history will not have a value on `sent_at` but I don't see this as blocking.